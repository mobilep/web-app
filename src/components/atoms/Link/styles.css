.Link {
	color: var(--white);
	font-size: 14px;
	font-weight: var(--normal);
	line-height: 1;
}

.Link:hover,
.Link:active {
	color: var(--white);
}
