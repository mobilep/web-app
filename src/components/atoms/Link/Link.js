import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Link as RouterLink } from 'react-router-dom';

import './styles.css';

const getClassName = (className) => classNames('Link', className);

const Link = ({ className, children, to, ...rest }) => (
	<RouterLink to={to} { ...rest } className={getClassName(className)}>
		{children}
	</RouterLink>
);

Link.propTypes = {
	children: PropTypes.oneOfType([PropTypes.string, PropTypes.element, PropTypes.array]),
	className: PropTypes.string,
	to: PropTypes.string.isRequired,
};

export default Link;
