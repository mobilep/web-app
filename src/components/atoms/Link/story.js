import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs/react';
import StoryRouter from 'storybook-react-router';

import Link from './Link';

const props = (({
	defaultHref = 'someLink',
} = {}) => ({
	to: text('to', defaultHref),
}));

storiesOf('1. Atoms / Link', module)
	.addDecorator(StoryRouter())
	.add('Link', () => <Link {...props()}>{text('text', 'someText')}</Link>);
