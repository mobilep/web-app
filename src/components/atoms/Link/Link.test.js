import React from 'react';
import { shallow } from 'enzyme';

import Link from './Link';

describe('Atom - Link', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<Link to='morrow'>
				Link text
			</Link>
		);
		expect(wrapper).toMatchSnapshot();
	});

});

