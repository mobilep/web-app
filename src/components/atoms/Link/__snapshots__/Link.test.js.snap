// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - Link Should render 1`] = `
<Link
  className="Link"
  replace={false}
  to="morrow"
>
  Link text
</Link>
`;
